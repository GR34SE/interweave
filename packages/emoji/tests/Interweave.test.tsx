import React from 'react';
import { render } from 'rut';
import Interweave, { Element } from 'interweave';
import { SOURCE_PROP } from 'interweave/lib/testUtils';
import EmojiMatcher from '../src/EmojiMatcher';

describe('Interweave (with emoji)', () => {
  it('renders emoji shortcode as unicode', () => {
    const { root } = render(
      <Interweave
        tagName="div"
        matchers={[new EmojiMatcher('emoji', { convertShortcode: true, renderUnicode: true })]}
        content="This has :cat: and :dog: shortcodes."
        emojiSource={SOURCE_PROP}
      />,
    );

    expect(root.findAt(Element, 'first')).toMatchSnapshot();
  });

  it('renders emoji unicode (literals) as unicode', () => {
    const { root } = render(
      <Interweave
        tagName="div"
        matchers={[new EmojiMatcher('emoji', { convertUnicode: true, renderUnicode: true })]}
        content="This has 🐈️ and 🐕️ shortcodes."
        emojiSource={SOURCE_PROP}
      />,
    );

    expect(root.findAt(Element, 'first')).toMatchSnapshot();
  });

  it('renders emoji unicode (escapes) as unicode', () => {
    const { root } = render(
      <Interweave
        tagName="div"
        matchers={[new EmojiMatcher('emoji', { convertUnicode: true, renderUnicode: true })]}
        content={'This has \uD83D\uDC31 and \uD83D\uDC36 shortcodes.'}
        emojiSource={SOURCE_PROP}
      />,
    );

    expect(root.findAt(Element, 'first')).toMatchSnapshot();
  });

  it('renders a single emoji enlarged', () => {
    const { root } = render(
      <Interweave
        tagName="div"
        matchers={[new EmojiMatcher('emoji', { convertUnicode: true, convertShortcode: true })]}
        content=":cat:"
        emojiSource={SOURCE_PROP}
      />,
    );

    expect(root.findAt(Element, 'first')).toMatchSnapshot();
  });
});
