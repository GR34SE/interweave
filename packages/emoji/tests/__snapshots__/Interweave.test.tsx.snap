// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Interweave (with emoji) renders a single emoji enlarged 1`] = `
<Element tagName="div">
  <div>
    <Emoji
      key="0"
      enlargeEmoji
      emojiLargeSize="3em"
      emojiPath="{{hexcode}}"
      emojiSize="1em"
      emojiSource={{ compact: false, locale: "en", version: "latest" }}
      emoticon=""
      hexcode="1F408"
      renderUnicode={false}
      shortcode=":cat:"
      unicode=""
    >
      <img
        alt="🐈️"
        aria-label="cat"
        data-hexcode="1F408"
        data-shortcodes=":cat:"
        data-unicode="🐈️"
        src="1F408"
        style={{ display: "inline-block", verticalAlign: "middle", width: "3em", height: "3em" }}
        title="cat"
      />
    </Emoji>
  </div>
</Element>
`;

exports[`Interweave (with emoji) renders emoji shortcode as unicode 1`] = `
<Element tagName="div">
  <div>
    This has 
    <Emoji
      key="0"
      renderUnicode
      emojiLargeSize="3em"
      emojiPath="{{hexcode}}"
      emojiSize="1em"
      emojiSource={{ compact: false, locale: "en", version: "latest" }}
      emoticon=""
      enlargeEmoji={false}
      hexcode="1F408"
      shortcode=":cat:"
      unicode=""
    >
      <span>🐈️</span>
    </Emoji>
     and 
    <Emoji
      key="1"
      renderUnicode
      emojiLargeSize="3em"
      emojiPath="{{hexcode}}"
      emojiSize="1em"
      emojiSource={{ compact: false, locale: "en", version: "latest" }}
      emoticon=""
      enlargeEmoji={false}
      hexcode="1F415"
      shortcode=":dog:"
      unicode=""
    >
      <span>🐕️</span>
    </Emoji>
     shortcodes.
  </div>
</Element>
`;

exports[`Interweave (with emoji) renders emoji unicode (escapes) as unicode 1`] = `
<Element tagName="div">
  <div>
    This has 
    <Emoji
      key="0"
      renderUnicode
      emojiLargeSize="3em"
      emojiPath="{{hexcode}}"
      emojiSize="1em"
      emojiSource={{ compact: false, locale: "en", version: "latest" }}
      emoticon=""
      enlargeEmoji={false}
      hexcode="1F431"
      shortcode=""
      unicode="🐱"
    >
      <span>🐱</span>
    </Emoji>
     and 
    <Emoji
      key="1"
      renderUnicode
      emojiLargeSize="3em"
      emojiPath="{{hexcode}}"
      emojiSize="1em"
      emojiSource={{ compact: false, locale: "en", version: "latest" }}
      emoticon=""
      enlargeEmoji={false}
      hexcode="1F436"
      shortcode=""
      unicode="🐶"
    >
      <span>🐶</span>
    </Emoji>
     shortcodes.
  </div>
</Element>
`;

exports[`Interweave (with emoji) renders emoji unicode (literals) as unicode 1`] = `
<Element tagName="div">
  <div>
    This has 
    <Emoji
      key="0"
      renderUnicode
      emojiLargeSize="3em"
      emojiPath="{{hexcode}}"
      emojiSize="1em"
      emojiSource={{ compact: false, locale: "en", version: "latest" }}
      emoticon=""
      enlargeEmoji={false}
      hexcode="1F408"
      shortcode=""
      unicode="🐈️"
    >
      <span>🐈️</span>
    </Emoji>
     and 
    <Emoji
      key="1"
      renderUnicode
      emojiLargeSize="3em"
      emojiPath="{{hexcode}}"
      emojiSize="1em"
      emojiSource={{ compact: false, locale: "en", version: "latest" }}
      emoticon=""
      enlargeEmoji={false}
      hexcode="1F415"
      shortcode=""
      unicode="🐕️"
    >
      <span>🐕️</span>
    </Emoji>
     shortcodes.
  </div>
</Element>
`;
